### DISCUSSION TITLE
Erroneous behaviour of background color style of a textbox even when its disabled state is toggled.

### DISCUSSION THREAD

**Discussion by AnuragBhatt** [unoplatform/uno#7787] [General]
I am trying to modify the isEnabled property of the textbox dynamically as shown in the code below: 
<TextBox Name="SearchTxt" KeyDown="SearchText_Keydown"
       IsEnabled="{x:Bind ViewModel.IsSearchEnabled, Mode=OneWay}"
Text="{x:Bind ViewModel.SearchText, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}">
                </TextBox>

However, it behaves quite strangely. 

ViewModel.IsSearchEnabled being a Boolean is initially false by default.
After a while in page load, I am setting it to true in View Model.
Now it is shows up in the UI as if it is still disabled (looking at background color).
 (Picture 1)
But if I try to hover and click on it, it then changes the color to default white (and actually I am able to click upon it, so that means IsEnabled has worked).
 
So this behaviour is only observed when I try to bind the “IsEnabled” attribute to some Viewmodel property.
If I try to hardcode IsEnabled as true or false, everything works just fine.

Style triggers won’t also work in UWP/WinUI world, but they used to work in WPF.
Not sure about VisualStateManager triggers.
Could somebody please point me how to get past this issue?
This is causing UI to not behave properly in many of the places.

code is shared at this location:
https://github.com/AnuragBhatt/Uno.Client

Steps to run the code:

1. download the code from repo
2. Make Uno.Client.Wasm as startup project.
3. Run the project.
4. Initially you will see the textbox disabled with "Initial Text" as content (try clicking on it to ensure that it is disabled)
5. Click on the button below, that will change the text of the text box and will enable it.
6. Problem found: the background color style of the textbox still gives the visual effect of a disabled textbox whereas it should be enabled and shown in white.
7. if you hover upon the textbox, it magically changes the background as desired.

**Comment by jeromelaban** [comment]
Thanks for the report! Indeed, you've found an visual state issue in TextBox! Let's track it here: https://github.com/unoplatform/uno/issues/7791

**Reply by AnuragBhatt** [reply]
Kudos to the UNO team, I can confirm this has been resolved now
